File: android/src/main/java/com/reactnativecblite/Args/IndexArgs.java
package com.reactnativecblite.Args;

import java.util.List;

public class IndexArgs {
    private String dbName;
    private String indexName;
    private boolean ignoreAccents;
    private String language;
    private List<String> indexExpressions;


    public IndexArgs() {}

    public void DeleteIndexArgs(String dbName,String indexName) {
        this.dbName = dbName;
        this.indexName = indexName;
    }

    public void ValueIndexArgs(String dbName,String indexName,List<String> indexExpressions) {
        this.dbName = dbName;
        this.indexName = indexName;
        this.indexExpressions = indexExpressions;
    }

    public void FTSIndexArgs(String databaseName, String indexName, boolean ignoreAccents, String language, List<String> indexExpressions) {

        this.dbName = databaseName;
        this.indexName = indexName;
        this.ignoreAccents = ignoreAccents;
        this.language = language;
        this.indexExpressions = indexExpressions;

    }




    public boolean isIgnoreAccents() {
        return ignoreAccents;
    }

    public void setIgnoreAccents(boolean ignoreAccents) {
        this.ignoreAccents = ignoreAccents;
    }

    public String getLanguage() {
        return language;
    }

    public void setLanguage(String language) {
        this.language = language;
    }



    public List<String> getIndexExpressions() {
        return indexExpressions;
    }

    public void setIndexExpressions(List<String> indexes) {
        this.indexExpressions = indexes;
    }

    public String getDbName() {
        return dbName;
    }

    public void setDbName(String dbName) {
        this.dbName = dbName;
    }

    public String getIndexName() {
        return indexName;
    }

    public void setIndexName(String indexName) {
        this.indexName = indexName;
    }
}

File: ios/Args/IndexArgs.swift
//
//  IndexArgs.swift
//  RNReactNativeCblite
//
//  Created by Umer on 05/12/2021.
//  Copyright © 2021 Facebook. All rights reserved.
//

//import UIKit

class IndexArgs: NSObject {
    private var _dbName:String?
    var dbName:String? {
        get { return _dbName }
        set { _dbName = newValue }
    }
    private var _indexName:String?
    var indexName:String? {
        get { return _indexName }
        set { _indexName = newValue }
    }
    private var _ignoreAccents:Bool?
    var ignoreAccents:Bool? {
        get { return _ignoreAccents }
        set { _ignoreAccents = newValue }
    }
    private var _language:String?
    var language:String? {
        get { return _language }
        set { _language = newValue }
    }
    private var _indexExpressions:[String]?
    var indexExpressions:[String]? {
        get { return _indexExpressions }
        set { _indexExpressions = newValue }
    }
    
    override init() {
        super.init()
    }
    
    func DeleteIndexArgs(dbName: String, indexName: String) {
        self.dbName = dbName
        self.indexName = indexName
    }
    
    func ValueIndexArgs(dbName: String, indexName: String, indexExpressions: [String]) {
        self.dbName = dbName
        self.indexName = indexName
        self.indexExpressions = indexExpressions
    }
    
    func FTSIndexArgs(dbName: String, indexName: String, ignoreAccents: Bool, language: String, indexExpressions: [String]) {
        self.dbName = dbName
        self.indexName = indexName
        self.ignoreAccents = ignoreAccents
        self.language = language
        self.indexExpressions = indexExpressions
    }
}
